hub/core/index/index.py

```diff
@@ -417,10 +417,23 @@ def to_json(self):
             v = e.value
             if isinstance(v, slice):
                 ret.append({"start": v.start, "stop": v.stop, "step": v.step})
+            elif isinstance(v, Iterable):
```

Comment:
```suggestion
            elif isinstance(v, Iterable) or callable(v):
```

Reply:
No, we do different things for Iterables and Callables. (`list(v)` vs `list(v())`)